zcash_primitives/src/zip32/fingerprint.rs

```diff
@@ -0,0 +1,53 @@
+//! Seed Fingerprints according to ZIP 32
+//!
+//! Implements section `Seed Fingerprints` of Shielded Hierarchical Deterministic Wallets (ZIP 32)
+//!
+//! [Section Seed Fingerprints]: https://zips.z.cash/zip-0032#seed-fingerprints
+
+use blake2b_simd::Params as Blake2bParams;
+
+pub const ZIP32_SEED_FP_PERSONALIZATION: &[u8; 16] = b"Zcash_HD_Seed_FP";
+
+pub struct SeedFingerprint([u8; 32]);
+
+impl SeedFingerprint {
+    pub fn from_seed(seed_bytes: &[u8]) -> SeedFingerprint {
+        let mut h = Blake2bParams::new()
+            .hash_length(32)
+            .personal(ZIP32_SEED_FP_PERSONALIZATION)
+            .to_state();
+        let seed_len_bytes = seed_bytes.len().to_le_bytes();
+
+        let concatenated_input = [&seed_len_bytes, seed_bytes].concat();
+
+        h.update(&concatenated_input);
```

Comment:
```suggestion
        h.update(seed_bytes.len().to_le_bytes());
        h.update(&seed_bytes);
```

Reply:
Oh I didn't know update would mean to concatenate.

I have a question about the implementation: is the `to_le_bytes()` correct? The ZIP says all bytes are `le` but IDK if seed bytes are already `le`.

Reply:
The [specification](https://zips.z.cash/zip-0032#seed-fingerprints) says to use a single-byte length, so it is not necessary to define a byte order.

In any case, `len()` returns a `usize`, and [usize::to_le_bytes](https://doc.rust-lang.org/std/primitive.usize.html#method.to_le_bytes) is not correct because, as the documentation says, it returns an array of length 2, 4 or 8 bytes depending on the target pointer size.

Reply:
```suggestion
        let seed_len = seed_bytes.len();
        assert!(32 <= seed_len && seed_len <= 252);
        h.update(&[u8::try_from(seed_len).unwrap()]);
        h.update(seed_bytes);
```